source.txt

```diff
@@ -506,6 +506,11 @@ Table of Contents
     SHOULD connect to the given provider account, as defined in Section
     10.
 
+    If the `remotestorage` field exists in the URL fragment, the application
+    SHOULD ignore any other parameters such as `access_token` or `state`, to
+    ensure compatibility with servers that implement older versions of this
+    specification.
+
```

Comment:
With the current phrasing, a draft-06 server could send the parameters
`remotestorage` and `access_token` to a draft-07 client. The draft-07 client
then could try to read `access_token` (instead of `remotestorage`), finds out
that the `state` parameter is missing/invalid and then discard the entire URL
fragment, or even show an error message.

What I actually wanted to express with this is a prioritization of
`remotestorage` parameter over `access_token`, `scope`, `state` etc.

On Sat, May 28, 2016 at 07:06:37AM -0700, Sebastian Kippe wrote:
> > @@ -506,6 +506,11 @@ Table of Contents
> >      SHOULD connect to the given provider account, as defined in Section
> >      10.
> >  
> > +    If the `remotestorage` field exists in the URL fragment, the application
> > +    SHOULD ignore any other parameters such as `access_token` or `state`, to
> > +    ensure compatibility with servers that implement older versions of this
> > +    specification.
> > +
> 
> I was thinking about this for a bit, and it's actually not clear to me how ignoring parameters ensures compatibility with older servers. I think it's the other way around: requiring/expecting them would break compatibility with *newer* servers, no?
> 
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/remotestorage/spec/pull/149/files/4abec030395702ee16df9549c0ace0d3ea2008c7#r64989358